src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,196 @@
+import java.util.Date;
+import java.text.SimpleDateFormat;
+
+public class Parser {
+    public static Integer parseInt(String str) {
+        try {
+            return Integer.parseInt(str);
+        } catch (NumberFormatException e) {
+            throw new CommandInvalidArgumentFormatException();
+        } catch (Exception e) {
+            throw e;
+        }
+    }
+
+    public static Date parseDate(String dateStr)
+            throws ParseException {
+        SimpleDateFormat format = new SimpleDateFormat("dd-MM-YYYY");
+        Date date = format.parse(dateStr);
+        return date;
+    }
+
+    /**
+     * The parser is able to parse input from file
+     */
+    public static ArrayList<Task> parseFile(String string)
+            throws StreamCorruptedException, ParseException {
+        Scanner scanner = new Scanner(string);
+        ArrayList tasks = new ArrayList<>();
+        while (scanner.hasNextLine()) {
+            String nextLine = scanner.nextLine();
+            Task task = Parser.parseFileLine(nextLine);
+            tasks.add(task);
+        }
+        return tasks;
+    }
+
+    /**
+     * Parses a single line of input from file
+     */
+    public static Task parseFileLine(String string)
+            throws StreamCorruptedException, ParseException {
+        String[] tokens = string.split(",");
+        String type = tokens[0];
+        switch (type) {
+            case "[T]": {
+                // Todo task
+
+                if (tokens.length <= 1 || tokens[2].equals("")) {
+                    throw new CommandMissingArgumentException();
+                }
+
+                // Create the task
+                return new TodoTask(tokens[1].equals("1"), tokens[2]);
+            }
+            case "[E]": {
+                // Events task
+
+                // Arguments
+                if (tokens.length <= 2
+                        || tokens[2].equals("")
+                        || tokens[3].equals("")) {
+                    throw new CommandMissingArgumentException();
+                }
+
+                // Otherwise create the task
+                return new EventTask(
+                        tokens[1].equals("1"),
+                        tokens[2],
+                        parseDate(tokens[3])
+                );
+            }
+            case "[D]": {
+                // Deadline task
+
+                // Arguments
+                if (tokens.length <= 2
+                        || tokens[2].equals("")
+                        || tokens[3].equals("")) {
+                    throw new CommandMissingArgumentException();
+                }
+
+                // Otherwise create the task
+                return new DeadlineTask(
+                        tokens[1].equals("1"),
+                        tokens[2],
+                        parseDate(tokens[3])
+                );
+            }
+            default: {
+                throw new StreamCorruptedException("File is corrupted");
+            }
```

Comment:
The coding style for this module suggests we should have no indentation for case clause 😕